lhotse/cut.py

```diff
@@ -1152,15 +1163,14 @@ def with_features_path_prefix(self, path: Pathlike) -> 'CutSet':
     def with_recording_path_prefix(self, path: Pathlike) -> 'CutSet':
         return CutSet.from_cuts(c.with_recording_path_prefix(path) for c in self)
 
-    def apply_supervision_modifications(self, process_supervision: Callable[[SupervisionSegment], None]) -> None:
+    def map_supervisions(self, transform_fn: Callable[[SupervisionSegment], SupervisionSegment]) -> 'CutSet':
```

Comment:
We'll need that method for `PaddingCut` and `MixedCut` too

Reply:
(for `PaddingCut` it can just return self, for `MixedCut` it should apply this method on all tracks and return a new `MixedCut`)

Reply:
`Cut`, `PaddingCut` and `MixedCut` are all derived from `CutUtilsMixin`, aren't them? So I implemented `map_supervisions` as a member of `CutUtilsMixin`.

Reply:
Emm.. yes, this method should perform different behaviors in these three classes. I'll add for them separately.

Reply:
Ha, I missed that the method is in `CutUtilsMixin`. Yeah, they'll need different behaviours - the cuts have definitely grown over time... 😀